packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -308,6 +309,23 @@ export interface AwsCustomResourceProps {
    * ID for the function's name. For more information, see Name Type.
    */
   readonly functionName?: string;
+
+  /**
+   * The vpc to provision the lambda function in.
+   *
+   * @default - the function is not provisioned inside a vpc.
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * Which subnets from the VPC to place the lambda function in.
+   *
+   * Only used if 'vpc' is supplied. Note: internet access for Lambdas
+   * requires a NAT gateway, so picking Public subnets is not allowed.
```

Comment:
Another thing that needs to be enforced in code. What happens if someone picks public subnets? Will this fail at deployment time or at synth?

Reply:
I believe synth time. https://github.com/aws/aws-cdk/blob/17f1ec881ba8fb300bd4cf8674a87640ab05c31a/packages/%40aws-cdk/aws-lambda/lib/function.ts#L1131-L1136

This doesn't allow for passing the `allowPublicSubnets` prop.